@emilio, `./mach build-geckolib` doesn't report any errors, but `./mach build` fails:
```
error[E0432]: unresolved import `values::specified::MozContextProperties`
  --> components/style/values/computed/svg.rs:16:9
   |
16 | pub use values::specified::MozContextProperties;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `MozContextProperties` in `values::specified`

error: aborting due to previous error

error: Could not compile `style`.
```
Not quite sure how to fix this, could you help?

--- a/components/style/properties/gecko.mako.rs
+++ b/components/style/properties/gecko.mako.rs
@@ -5257,16 +5257,16 @@ clip-path
 
         self.gecko.mContextPropsBits = 0;
         for (gecko, servo) in self.gecko.mContextProps.iter_mut().zip(v) {
-            if servo.0 == atom!("fill") {
+            if (servo.0).0 == atom!("fill") {
                 self.gecko.mContextPropsBits |= structs::NS_STYLE_CONTEXT_PROPERTY_FILL as u8;
-            } else if servo.0 == atom!("stroke") {
+            } else if (servo.0).0 == atom!("stroke") {
                 self.gecko.mContextPropsBits |= structs::NS_STYLE_CONTEXT_PROPERTY_STROKE as u8;
-            } else if servo.0 == atom!("fill-opacity") {
+            } else if (servo.0).0 == atom!("fill-opacity") {
                 self.gecko.mContextPropsBits |= structs::NS_STYLE_CONTEXT_PROPERTY_FILL_OPACITY as u8;
-            } else if servo.0 == atom!("stroke-opacity") {
+            } else if (servo.0).0 == atom!("stroke-opacity") {
                 self.gecko.mContextPropsBits |= structs::NS_STYLE_CONTEXT_PROPERTY_STROKE_OPACITY as u8;
             }
-            gecko.mRawPtr = servo.0.into_addrefed();
+            gecko.mRawPtr = (servo.0).0.into_addrefed();
         }
     }
 

--- a/components/style/properties/longhand/inherited_svg.mako.rs
+++ b/components/style/properties/longhand/inherited_svg.mako.rs
@@ -136,24 +136,11 @@ ${helpers.predefined_type("paint-order", "SVGPaintOrder", "computed::SVGPaintOrd
                           gecko_pref="svg.paint-order.enabled",
                           spec="https://www.w3.org/TR/SVG2/painting.html#PaintOrder")}
 
-<%helpers:vector_longhand name="-moz-context-properties"
-                   animation_value_type="none"
-                   products="gecko"
-                   spec="Nonstandard (Internal-only)"
-                   allow_empty="True">
-    use values::CustomIdent;
-
-    pub type SpecifiedValue = CustomIdent;
-
-    pub mod computed_value {
-        pub type T = super::SpecifiedValue;
-    }
-
-
-    pub fn parse<'i, 't>(_context: &ParserContext, input: &mut Parser<'i, 't>)
-                         -> Result<SpecifiedValue, ParseError<'i>> {
-        let location = input.current_source_location();
-        let i = input.expect_ident()?;
-        CustomIdent::from_ident(location, i, &["all", "none", "auto"])
-    }
-</%helpers:vector_longhand>
+${helpers.predefined_type("-moz-context-properties",
+                          "MozContextProperties",
+                          initial_value=None,
+                          vector=True,
+                          animation_value_type="none",
+                          products="gecko",
+                          spec="Nonstandard (https://developer.mozilla.org/en-US/docs/Web/CSS/-moz-context-properties)",
+                          allow_empty=True)}

--- a/components/style/values/computed/mod.rs
+++ b/components/style/values/computed/mod.rs
@@ -62,6 +62,7 @@ pub use self::percentage::Percentage;
 pub use self::position::{Position, GridAutoFlow, GridTemplateAreas};
 pub use self::svg::{SVGLength, SVGOpacity, SVGPaint, SVGPaintKind};
 pub use self::svg::{SVGPaintOrder, SVGStrokeDashArray, SVGWidth};
+pub use self::svg::MozContextProperties;
 pub use self::table::XSpan;
 pub use self::text::{InitialLetter, LetterSpacing, LineHeight, TextAlign, TextOverflow, WordSpacing};
 pub use self::time::Time;

--- a/components/style/values/computed/svg.rs
+++ b/components/style/values/computed/svg.rs
@@ -13,6 +13,8 @@ use values::generics::svg as generic;
 
 pub use values::specified::SVGPaintOrder;
 
+pub use values::specified::MozContextProperties;
+
 /// Computed SVG Paint value
 pub type SVGPaint = generic::SVGPaint<RGBA, ComputedUrl>;
 /// Computed SVG Paint Kind value

--- a/components/style/values/specified/mod.rs
+++ b/components/style/values/specified/mod.rs
@@ -58,6 +58,7 @@ pub use self::percentage::Percentage;
 pub use self::position::{Position, PositionComponent, GridAutoFlow, GridTemplateAreas};
 pub use self::svg::{SVGLength, SVGOpacity, SVGPaint, SVGPaintKind};
 pub use self::svg::{SVGPaintOrder, SVGStrokeDashArray, SVGWidth};
+pub use self::svg::MozContextProperties;
 pub use self::table::XSpan;
 pub use self::text::{InitialLetter, LetterSpacing, LineHeight, TextDecorationLine};
 pub use self::text::{TextAlign, TextAlignKeyword, TextOverflow, WordSpacing};

--- a/components/style/values/specified/svg.rs
+++ b/components/style/values/specified/svg.rs
@@ -264,7 +264,8 @@ impl ToCss for SVGPaintOrder {
 
 /// Specified MozContextProperties value.
 /// Nonstandard (https://developer.mozilla.org/en-US/docs/Web/CSS/-moz-context-properties)
-pub type MozContextProperties = CustomIdent;
+#[derive(Clone, Debug, MallocSizeOf, PartialEq, ToComputedValue, ToCss)]
+pub struct MozContextProperties(pub CustomIdent);
 
 impl Parse for MozContextProperties {
     fn parse<'i, 't>(
@@ -273,6 +274,6 @@ impl Parse for MozContextProperties {
     ) -> Result<MozContextProperties, ParseError<'i>> {
         let location = input.current_source_location();
         let i = input.expect_ident()?;
-        CustomIdent::from_ident(location, i, &["all", "none", "auto"])
+        Ok(MozContextProperties(CustomIdent::from_ident(location, i, &["all", "none", "auto"])?))
     }
 }

--- a/components/style/values/specified/svg.rs
+++ b/components/style/values/specified/svg.rs
@@ -8,6 +8,7 @@ use cssparser::Parser;
 use parser::{Parse, ParserContext};
 use std::fmt;
 use style_traits::{CommaWithSpace, ParseError, Separator, StyleParseErrorKind, ToCss};
+use values::CustomIdent;
 use values::generics::svg as generic;
 use values::specified::{LengthOrPercentage, NonNegativeLengthOrPercentage, NonNegativeNumber};
 use values::specified::{Number, Opacity, SpecifiedUrl};
@@ -258,5 +259,20 @@ impl ToCss for SVGPaintOrder {
             self.order_at(pos).to_css(dest)?;
         }
         Ok(())
+    }
+}
+
+/// Specified MozContextProperties value.
+/// Nonstandard (https://developer.mozilla.org/en-US/docs/Web/CSS/-moz-context-properties)
+pub type MozContextProperties = CustomIdent;
+
+impl Parse for MozContextProperties {
+    fn parse<'i, 't>(
+        _context: &ParserContext,
+        input: &mut Parser<'i, 't>
+    ) -> Result<MozContextProperties, ParseError<'i>> {
+        let location = input.current_source_location();
+        let i = input.expect_ident()?;
+        CustomIdent::from_ident(location, i, &["all", "none", "auto"])
     }
 }

